### ISSUE TITLE
Buckets replication metrics

### ISSUE THREAD

**Issue by maxkushnir** [PureStorage-OpenConnect/pure-fb-openmetrics-exporter#84]
We are requesting the addition of new metrics to track the replication of buckets. The proposed metrics would offer detailed insights on a per-replicated bucket basis and also provide general replication metrics at the device level.

### Proposed Metrics:
**Per-replicated bucket:**

- _purefb_buckets_replica_status_: Status of replication for each bucket.
- _purefb_buckets_replica_backlog_size_: Size of the backlog for each replicated bucket.
- _purefb_buckets_replica_lag_: Replication lag per bucket.
- _purefb_buckets_replica_operations{type="put|delete|other"}_: Number of operations per bucket, categorized by type.

**Labels for each metric:**
- _bucket_name_: Name of the replicated bucket.
- _remote_host_: Hostname of the remote destination.

**General replication metrics:**
- _purefb_buckets_replica_backlog_size_total_: Total backlog size across all replicated buckets.
- _purefb_buckets_replica_operations_total{type="put|delete|other"}_: Total replication operations, categorized by type.

These metrics will greatly improve visibility into the replication status and performance of buckets.

Thank you for considering this request!

![image-2024-10-08-17-14-42-903](https://github.com/user-attachments/assets/db19ecb3-c9a3-4467-9f74-5d69a6f4d62b)

**Comment by chrroberts-pure** [comment]
Hi! Yes, I agree these metrics would be a great addition! Thank you for the feedback, i will keep this issue open as we develop this

**Comment by LiginGeorge** [comment]
Hi,
Can you please update the progress of this request. Is there any way we can configure mail notification via any API call

**Comment by chrroberts-pure** [comment]
Yes, this feature is still in our pipeline for development. Thank you.

**Comment by chrroberts-pure** [comment]
Hi! I'm starting to work on this. See my work in 

https://github.com/PureStorage-OpenConnect/pure-fb-openmetrics-exporter/compare/main...84-buckets-replication-metrics

So far I've implemented the rest client, and tested with this metric. It's data model is not finalized yet.

````
# HELP purefb_buckets_replica_lag_msec FlashBlade duration in milliseconds that represents how far behind the replication target is from the source
# TYPE purefb_buckets_replica_lag_msec gauge
purefb_buckets_replica_lag_msec{direction="outbound",local_bucket_name="test-repl-bucket",remote_bucket_name="test-repl-bucket",remote_name="FlashBlade04",status="replicating"} 28143
````

**Comment by assaffeuerstein** [comment]
Hi, any news regarding this issue?

**Comment by maxkushnir** [comment]
<img width="360" height="360" alt="Image" src="https://github.com/user-attachments/assets/6d6d6ffc-b296-4ec6-a775-35b62e9cc23d" />

Let's it happen in 2026 🤞

**Comment by assaffeuerstein** [comment]
https://github.com/PureStorage-OpenConnect/pure-fb-openmetrics-exporter/pull/128
